### PR TITLE
PXP-595: integrate github pipeline while fetching builds 

### PR DESCRIPTION
**Pull request by mfauzaan**
This PR updates the logic of existing command `wukong pipeline list` and `wukong deployment execute`.
New logic will fetch from github pipelines and if there is no pipeline found it fallbacks to fetch from jenkins to support the interim migration period 

## Summary

Ticket: [https://mindvalley.atlassian.net/browse/PXP-595](https://mindvalley.atlassian.net/browse/PXP-595)

## What's Changed

- [x]  Sync graphql schema with new changes
- [x] Update fetch pipeline to fetch from github pipelines

### CODE REVIEW

File: src/commands/deployment/execute.rs
Comment (jk-gan) [inline src/commands/deployment/execute.rs:914]: Perhaps the name should be changed? since now it is no longer only for Jenkins.

File: src/commands/deployment/execute.rs
Comment (jk-gan) [inline src/commands/deployment/execute.rs:399]: The error from `get_github_cd_pipeline` is not handled.

File: src/commands/deployment/execute.rs
Comment (jk-gan) [inline src/commands/deployment/execute.rs:400]: while the code is valid, but I would prefer to use `if let Ok(github_cd_pipeline_cd) = github_cd_pipeline { ... }` so we don't have so much unwrap.

File: src/commands/deployment/execute.rs
Comment (jk-gan) [inline src/commands/deployment/execute.rs:787]: I think we could just pass the builds to the function, so we don't need to check the pipeline type. (Provided `github_builds` and `jenkins_builds` are the same type.)

File: src/graphql/github_pipeline.rs
Comment (jk-gan) [inline src/graphql/github_pipeline.rs:70]: We have error code for this one. Please check the `error.rs` file.

**Review by jk-gan — CHANGES_REQUESTED** [review]

File: src/commands/deployment/execute.rs
Comment (mfauzaan) [reply src/commands/deployment/execute.rs]: Renamed `JenkinsBuild` to `CdPipelineBuild`

File: src/commands/deployment/execute.rs
Comment (mfauzaan) [reply src/commands/deployment/execute.rs:399]: Its intentional, reason being is that when github endpoint fails to get the pipeline weather pipeline is not setup/not_found we fallback and fetch from jenkins.

File: src/commands/deployment/execute.rs
Comment (mfauzaan) [reply src/commands/deployment/execute.rs]: Thanks, this helped me to look into this in a diff angle. now code looks much cleaner.

Still i kept the filteration within the func to reduce the logic in the main function. Main func has alot of code already.  :')

File: src/commands/deployment/execute.rs
Comment (jk-gan) [reply src/commands/deployment/execute.rs:399]: are we want to fallback to jenkins pipeline for any error message from github endpoint?

**Comment by sauron-droid** [comment]
PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/138#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cli/src/commands/deployment/execute.rs
Comment (jk-gan) [inline cli/src/commands/deployment/execute.rs:115]: I'm thinking how about just `builds`? Since we only use either `jenkins_builds` or `github_builds` at one time. What do you think?

**Review by jk-gan — DISMISSED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/138#pullrequestreview-1608854367" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/138#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/138#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/138#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jk-gan — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/138#pullrequestreview-1610209288" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/138#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
